### ISSUE TITLE
variables and taxonomy in our ro-crates

### ISSUE THREAD

**Issue by kmexter** [emo-bon/rocrate-metadata-generator-action#4]
For the logsheet ro-crates we have the total temporal and the spatial coverage listed. For exporting our datasets to blue cloud and fair ease, we need to include also the list of variables (measurements), instruments, and taxonomy. These are then used as search terms in the (I)DDAS

The best is to put these info in the ro-crates, and we can then figure out later how to get them into FE and BC. 

What we need to decide is 
- which measurements to include (all, only mandatory, any of those on the "sampling" tab also (this will be relevant to the logsheet ro-crates)
- how much taxonomy to include (this will be relevant to the MGF ro-crates)

**Comment by cpavloud** [comment]
I would say all of the measurements. 

For the taxonomy question, do you mean if we will keep e.g. only the phylum level or all the levels?

**Comment by cpavloud** [comment]
So you say that since we will be pointing to the data, we don't nee to have all the data as triples. 

Ok, this seems reasonable. 
Although we should/could give it a try with the NCBI taxonomy too, right?
WoRMS is good for some taxa but we cannot rely to it for everything (unfortunately), so we need to have working options available (I guess).

**Comment by kmexter** [comment]
no, that is not what I meant. I means that we need in the metadata a summary of the taxonomic classifications that is in the data. the question is: how to make this summary? @cedricdcc : your code can in principle do this on NCBI as well as on worms, but how much work would that involve - maybe in January (Cedric is on vacation already, I think) we can look at this?
But meanwhile, perhaps @cymon or @cpavloud (or anyone else who runs MGF) can have a look at one of the MFG outputs and see (1) how many different unique names (at the most specific level) are there and (2) if that number > 100 then if we were to go always only to the genus level, how many unique names would there be?

**Comment by cymon** [comment]
> But meanwhile, perhaps @cymon or @cpavloud (or anyone else who runs MGF) can have a look at one of the MFG outputs and see (1) how many different unique names (at the most specific level) are there and (2) if that number > 100 then if we were to go always only to the genus level, how many unique names would there be?

Some number for SEDIMENTS (which then to be larger than FILTERS so these are upper bounds):

- SSU
- Species: mean 38
- Genera: mean 207

- LSU
- Species: 119
- Genera: 288

As said previous, I think mentioning some identified taxa and not others, might lead people to believe that a non-listed taxon is not present. If we dont think that is a problem, it might be worth listing the 10 most abundant taxa, but that could be very similar from sample to sample in the same environment, so I'm not sure what that is worth either...

**Comment by cpavloud** [comment]
If BlueCloud makes changes to accommodate EurOBIS, we could use the higher hierarchies that are used for the EurOBIS datasets.

I think it would also be beneficial to list the most abundant taxa, as Cymon says, especially if we can make a note somewhere/somehow that there are more species in the RO-crates (and not just those few abundant ones). I am thinking from the user perspective here, it is something that could be useful to have.

**Comment by kmexter** [comment]
Bear in mind that the reason to add this information is for the automated level 1 or level 2 search that blue cloud's DDAS does now. you type in something and you get all datasets that have said that they have that "something". so while notes are useful, it will not help unless the person has managed to find that dataset, via whatever the "something" is that they search on, in the first place
From a coding POV, it is a lot easier to just include all species (or genus, or whatever the final name is) than it is to include some summary of them. These information will not be added by hand, eh!

**Comment by kmexter** [comment]
eh?

**Comment by cpavloud** [comment]
We currently have X number of RO-crates to include in the DDAS with Y number of unique species that we are thinking of adding in the metadata search. 
What happens if we have future RO-crates to include in the DDAS with different species names? How will those new species be added in the metadata search?

**Comment by kmexter** [comment]
Ah. The metadata that are unique to each run of MGF - found by each run of MGF - are included in the ro-crate-metadata.json file of that "dataset". so if we have X datasets, from X runs of MFG, with X ro-crates, then there will be X metadata records that DDAS will ingest, each with their own lists of species found

**Comment by kmexter** [comment]
I think that this discussion can be picked up after we have a first set of MGF data in BC: we can decide then if we want to include these measurments, taxonomic, and associatted summary information in our records
